extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -119,16 +120,22 @@ public void handle(AsyncResult<OAuth2Auth> event) {
 
                 auth = cf.join();
                 break;
-            } catch (OIDCException ex) {
-                if (i + 1 < connectionRetryCount) {
-                    try {
-                        Thread.sleep(2000);
-                    } catch (InterruptedException iex) {
-                        // continue connecting
+            } catch (Throwable throwable) {
+                while (throwable instanceof CompletionException && throwable.getCause() != null) {
+                    throwable = throwable.getCause();
+                }
+                if (throwable instanceof OIDCException) {
+                    if (i + 1 < connectionRetryCount) {
+                        try {
+                            Thread.sleep(2000);
```

Comment:
Can this be called in an IO thread?

Reply:
Hi Stuart @stuartwdouglas Good question, I don't know, do you mean avoid using the future at all ? It just joins till the OIDC server has been contacted, may be it is not needed. This is done at the application start up, no client requests are involved.

Reply:
@stuartwdouglas `runAsync` I guess, let me try, though I'd not've guessed it implies a more optimal thread consumption in this case :-)

Reply:
@stuartwdouglas Should be better now

Reply:
what I was actually getting at was should we change this to register a timer and callback, rather than blocking a thread for 2s?

Reply:
@stuartwdouglas Callback to what condition though ? Vertx tells us OIDC is not contactable. It won't let us save 1 sec out of those 2 secs by calling back when OIDC becomes contactable unless this timer you are proposing will be pinging OIDC every 100 milliseconds. Is it what you suggest be done here ? I'm not sure it is really worth it, we are not dealing with the actual client request here. May be just reducing a wait period from 2 secs to 200 milliseconds is a simpler version of the timer/callback approach ?